### ISSUE TITLE
Error generating report for only some samples

### ISSUE THREAD

**Issue by AlinaO311** [ConesaLab/SQANTI3#94]
Hi,

I have two separate collections of samples from collapsed reads and low quality reads. I can use the  qc script perfectly fine with both collections, however, the low quality collection fails to generate a report for the filtering script with this error:

Warning message:
`guides(<scale> = FALSE)` is deprecated. Please use `guides(<scale> = "none")` instead.
Warning message:
`guides(<scale> = FALSE)` is deprecated. Please use `guides(<scale> = "none")` instead.
Warning message:
`guides(<scale> = FALSE)` is deprecated. Please use `guides(<scale> = "none")` instead.
Warning message:
`guides(<scale> = FALSE)` is deprecated. Please use `guides(<scale> = "none")` instead.
Warning messages:
1: `guides(<scale> = FALSE)` is deprecated. Please use `guides(<scale> = "none")` instead.
2: `guides(<scale> = FALSE)` is deprecated. Please use `guides(<scale> = "none")` instead.
Warning message:
`guides(<scale> = FALSE)` is deprecated. Please use `guides(<scale> = "none")` instead.
Warning message:
`guides(<scale> = FALSE)` is deprecated. Please use `guides(<scale> = "none")` instead.
`summarise()` has grouped output by 'lenCat'. You can override using the `.groups` argument.
Warning message:
`guides(<scale> = FALSE)` is deprecated. Please use `guides(<scale> = "none")` instead.
Warning message:
`guides(<scale> = FALSE)` is deprecated. Please use `guides(<scale> = "none")` instead.
Warning messages:
1: `guides(<scale> = FALSE)` is deprecated. Please use `guides(<scale> = "none")` instead.
2: `guides(<scale> = FALSE)` is deprecated. Please use `guides(<scale> = "none")` instead.
Error in `[.data.frame`(t3.a.SJ, , c(1, 5, 6)) :
  undefined columns selected
Calls: rbind ... standardGeneric -> eval -> eval -> eval -> [ -> [.data.frame
Execution halted
Traceback (most recent call last):
  File "/jumbo/WorkingDir/B21-010/Intermediate/SQANTI3-4.2/sqanti3_RulesFilter.py", line 227, in <module>
    main()
  File "/jumbo/WorkingDir/B21-010/Intermediate/SQANTI3-4.2/sqanti3_RulesFilter.py", line 222, in main
    sqanti_filter_lite(args)
  File "/jumbo/WorkingDir/B21-010/Intermediate/SQANTI3-4.2/sqanti3_RulesFilter.py", line 165, in sqanti_filter_lite
    if subprocess.check_call(cmd, shell=True)!=0:
  File "/home/xoroal/miniconda3/envs/SQANTI3.env/lib/python3.9/subprocess.py", line 373, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '/apps/bio/software/R/4.0.1/R-4.0.1/bin/Rscript /jumbo/WorkingDir/B21-010/Intermediate/SQANTI3-4.2/utilities//SQANTI3_report.R /jumbo/WorkingDir/B21-010/Intermediate/lq_isoforms/dedup.5merge.CJ30_lq.collapsed_classification.filtered_lite_classification.txt /jumbo/WorkingDir/B21-010/Intermediate/lq_isoforms/dedup.5merge.CJ30_lq.collapsed_classification.filtered_lite_junctions.txt mock /jumbo/WorkingDir/B21-010/Intermediate/SQANTI3-4.2/utilities/ False pdf' returned non-zero exit status 1.

Any idea about why I might get this error and why it is happening for only some samples? Thank you!

**Comment by aarzalluz** [comment]
Hi @AlinaO311 

We just released a new SQANTI3 version (v5.1) and we would like to confirm whether this issue is still occurring in order to fix it. Could you [download](https://github.com/ConesaLab/SQANTI3/wiki/Dependencies-and-installation#1-downloading-sqanti3) the new SQ3 and try it out?

Best,

Ángeles

**Comment by AlinaO311** [comment]
Hi @aarzalluz sorry the the delayed response. I had to remove and reinstall the dependencies. I am having trouble reinstalling the newest version of SQANTI3 though, not sure if you have any insight on this? I created a conda env with python 3.7 on a linux system and when I go to install from the .yml I get the following error: 
`ERROR conda.core.link:_execute(732): An error occurred while installing package 'bioconda::bioconductor-genomeinfodbdata-1.2.7-r41hdfd78af_1'.
Rolling back transaction: done
class: LinkError
message:
post-link script failed for package bioconda::bioconductor-genomeinfodbdata-1.2.7-r41hdfd78af_1
location of failed script: $CONDAENV/envs/SQANTI3.env/bin/.bioconductor-genomeinfodbdata-post-link.sh
==> script messages <==
<None>
==> script output <==
stdout: $CONDAENV/envs/SQANTI3.env/share/bioconductor-genomeinfodbdata-1.2.7-1/GenomeInfoDbData_1.2.7.tar.gz: FAILED
$CONDAENVenvs/SQANTI3.env/share/bioconductor-genomeinfodbdata-1.2.7-1/GenomeInfoDbData_1.2.7.tar.gz: FAILED
$CONDAENV/envs/SQANTI3.env/share/bioconductor-genomeinfodbdata-1.2.7-1/GenomeInfoDbData_1.2.7.tar.gz: FAILED
ERROR: post-link.sh was unable to download any of the following URLs with the md5sum 74c82f26111062a9ceb3c5331088cd56:
https://bioconductor.org/packages/3.14/data/annotation/src/contrib/GenomeInfoDbData_1.2.7.tar.gz
https://bioarchive.galaxyproject.org/GenomeInfoDbData_1.2.7.tar.gz
https://depot.galaxyproject.org/software/bioconductor-genomeinfodbdata/bioconductor-genomeinfodbdata_1.2.7_src_all.tar.gz
`
I've done some googling about what the problem could be but didn't have this issue with the past versions of Squanti3 and the previous versions were installed in this same way so I am not sure if this is just an issue with my system or not.

Thanks!

**Comment by aarzalluz** [comment]
Hi @AlinaO311 -I am not sure about what could be going on here, because the R package `genomeinfodbdata` is not one of SQ3's listed dependencies, so it must be required by one of the packages that we install in the environment. Are you removing the old conda environment and creating it again from scratch?

Perhaps you could have a look at [this issue](https://github.com/bioconda/bioconda-recipes/issues/13846) from the package repository in which a similar error is reported. It seems to be a conda error, rather than a problem with the package itself.

Ángeles

**Comment by AlinaO311** [comment]
Hi @aarzalluz ! There have been a ton of changes to Sqanti since I posted this question (and I also got a better understanding of the use if this tool with some of the others mentioned in the documentation, like Cupcake) . For example, it seems the input for Sqanti at the filtering level has changed so much that the output does not include filtered gff files even. So with that being said, I believe that the issue has been resolved. :) Really appreciate the work you all put into this!
